### PR TITLE
USWDS-site data-visualizations: 10x adding guidance and resources for accessible data visualizations

### PR DESCRIPTION
**Pull request by mike-stern**
[Preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/10x-data-viz/components/data-visualizations/) → 

## USWDS-site data-visualizations: 10x adding guidance and resources for accessible data visualizations
This content is part of the closeout for two 10x phase 2 projects: 

- [Reusable Patterns for Data Visualization](https://trello.com/c/wqJc9HWr)
- [Accessibility for Data Visualizations](https://trello.com/c/OOOres7a)

While the phase II closeout report does not recommend further investment, the content in this PR is intended to further the conversation and give a start to data visualizations as part of the USWDS. The link to the closeout document will be added below when it reviewed by 10x. The work so far was done on a private fork, with @jaredcunha contributing the line and bar chart examples as well as a fair amount of the content. 

## Additional information
* Phase II closeout report coming soon
* copy editing needed
* Review will remain open until at least **May 11th**
* Input welcome, any changes that can't be made at this time will have issues created to track for future enhancement

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.

getting an error running locally on master and branch, seems ok in [circleCI](https://circleci.com/gh/uswds/uswds-site/3005)
```
> @ crawl /Users/xx/dev/repos/uswds-public/uswds-site
> node config/crawl.js

Error: duplicate id attrs found at /form-controls/07-range-input/:
  range-slider
1 error(s) and 0 warning(s) found.
```
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Comment by jstrothman** [comment]
This is awesome to see! I’m pretty new around here, but used the USWDS sketch files on a few project before joining 18F, and was very interested in the addition of basic data visualization guidance in the system. 
I particularly appreciate the encouragement to include explanation below chart titles.  
I have a few suggestions for future iterations:

1. The Medium article on choosing a color palette is very helpful. I'd love to see USWDS offer examples of divergent palettes that are accessible _and look good_ even with a lot of colors (e.g. 7).
2. Chart axis titles are an important aid to help people understand the data in charts. In the Roosevelts example, the explanatory text is important for the scale of relative popularity. The dates on the x axis can reasonably be interpreted as dates and 2020 is in the chart title. Nevertheless, for people who read the title and axis labels first, or skip the explanation entirely, it would be helpful to have a y-axis title of 'Relative Popularity of Search Terms' and for the x-axis of '2020'.
3. It would be nice to have an alternate version of the legend on the right side of the chart, with a label above the legend, e.g. 'Search Terms' in the Roosevelt example. The right side option could move below the chart on small screen sizes. This option would be useful in cases when the legend labels are longer, so they can wrap without pushing the chart display further from the title and explanation, and keeping the legend visible even when the chart is tall. 
4. Perhaps make the chart example and guidance use varied data point shapes to help people distinguish which legend item applies to which line. The documentation mentions doing this when color contrast is not sufficient. However, it is very difficult to ensure sufficient difference across color deficiency types for a chart with more than 3 colors/lines.

**Comment by Jacklynn** [comment]
I usually search for a real example when I'm looking for help so I was glad to see the charts at the end. 

Maybe this doesn't match the pattern for how you're documenting things but, could you also call out the how the chart might interacted with for someone using a screen reader? Or resources? A question I've see a couple of times is how to label a complex graphic so having a good example could go a long way.

**Comment by jaredcunha** [comment]
Adding in some of @thisisdano's comments from earlier work here:

Added some responses to these. 

---

> 1. In the line chart example, I find it a little confusing that the tick marks on the x-axis define the boundaries of the day/label — not the center — while the marks on the y-axis are centered and aligned as expected. I don't think the x- and y-axes should use different methods for marks, and I think we should we rebuild the graph so the x-axis markers align to both the center of the labels and the dots on the graph.

 Regarding the "National Parks" appearing in quotes, this was how the NPS website conveyed the designation. As much as I love the national parks, I've never been 100% how to describe that I'm visiting a "National Park" and not just any old national park in the in the national parks system.

    10 Most Visited "National Parks" in 2019
    Out of the 419 parks in the National Park System in 2019, there were 62 with the designation "national park" as part of their official name.

What are your thoughts? Is it worth articulating in this example?

---

> 2. In the bar chart example, we can remove the quotes from "National Park" in the title (and use sentence case). Also, the summary doesn't actually summarize the findings in the graph — we should probably call out something like Great Smoky Mountain is the most popular National Park by a factor of two.

For the tick marks, this is how Illustrator builds the chart by default. I'm not particularly thrown off by the disconnect in how they're used between X and Y axis, but I'm not seeing any benefit they're providing to the x-axis. I can take them out.

Also, in respect to calling out that Great Smokey Mountain National Park is visited more by a factor of two… that could be a very different type of editorializing than saying, "Here's the top 5." The former would be used to describe a surprising statistic or would demonstrate the sheer popularity of Great Smokey Mountain National Park, while the latter would take a more objective stance.

---
> 3. Regarding the summary, I think it's still a bit vague in the guidance that we recommend including a findings summary in the body of the visualization. I guess it's summarized in Clarity of intent but it could be more... clear.

File: _components/data-visualizations.md [hunk 1]
```diff
@@ -260,22 +270,37 @@ subnav:
   </table>
 
 
-  <p id="source_line">Source: <a href="https://trends.google.com/trends/explore?date=2020-03-01%202020-03-07&geo=US&q=%2Fm%2F07hyk,%2Fm%2F02yy8">Google Trends</a></p>
-</div>
+    <p id="source_line">Source: <a href="https://trends.google.com/trends/explore?date=2020-03-01%202020-03-07&geo=US&q=%2Fm%2F07hyk,%2Fm%2F02yy8">Google Trends</a></p>
+  </div>
+
+  <h3>What we with our example</h3>
```
Comment (mike-stern) [inline _components/data-visualizations.md:276]: small typo?

File: _components/data-visualizations.md [hunk 2]
```diff
@@ -406,15 +431,15 @@ subnav:
...
-  <h3>Guidance</h3>
-    <h4> In addition to the general guidance above, the following bar chart guidance applies:</h4>
-    <ul>
-      <li>When displaying multi-variant data: </li>
-        <ul>
-          <li>It is important to use discrete, high contrast colors or textured fill. These accomodateion allow for various levels of visual impairments to differentiate between variables. </li>
-          <li>The use of stacked bar charts for multi-variant data can be used if it is important to communicate the sum of the dataset.</li>
-        </ul>
-    </ul>
+  <h3>What we with our example</h3>
```
Comment (mike-stern) [inline _components/data-visualizations.md:434]: small typo?

**Comment by thisisdano** [comment]
@mike-stern @jaredcunha I made a number of edits for clarity and style, and did some minor formatting to clean up the spacing. Give it a look and make sure I didn't edit away anything important.

**Comment by mike-stern** [comment]
> @mike-stern @jaredcunha I made a number of edits for clarity and style, and did some minor formatting to clean up the spacing. Give it a look and make sure I didn't edit away anything important.

The changes look good to me. Thank you for taking the time to do the edits. 👍

**Comment by jaredcunha** [comment]
@thisisdano This looks great. We have one bullet point about keyboard traps, and I'm wondering since we decided not to make the data points inside the chart focusable if we should remove it. Feels like it might be a little out of place there now.

**Comment by thisisdano** [comment]
Thanks to everyone for contributing to this project — I think it'll be the start of even more useful and practical work around dataviz, and I'm glad we were able to get the ball rolling.

**Pull request merged**